react/features/base/settings/functions.any.js

```diff
@@ -256,3 +257,23 @@ export function shouldHideShareAudioHelper(state: Object): boolean {
 
     return state['features/base/settings'].hideShareAudioHelper;
 }
+
+/**
+ * Whether we should hide self view.
+ *
+ * @param {Object} state - Redux state.
+ * @returns {boolean}
+ */
+export function shouldHideSelfView(state: Object) {
+    return getParticipantCount(state) === 1 ? false : getHideSelfView(state);
+}
+
+/**
+ * Gets the disable self view setting.
+ *
+ * @param {Object} state - Redux state.
+ * @returns {boolean}
+ */
+export function getHideSelfView(state: Object) {
+    return state['features/base/config'].disableSelfView || state['features/base/settings'].disableSelfView;
```

Comment:
If the config is set to true, there's no way for the user to see the self-view, even if they try to. The user will see the toggle in settings but it won't do anything.
This is why when the config was loaded we updated the settings, so there would be only one option to check.

Reply:
Hum then, maybe when it is controlled from outside we can hide the setting from UI.

Reply:
Sounds good to me